cmake/CompilerChecks.cmake

```diff
@@ -32,7 +32,10 @@ function(use_libcxx out)
 endfunction()
 
 function(static_link_libcxx out)
-  if(APPLE)
+  if(USE_TSAN)
```

Comment:
Introducing this means we won't use `lld` for clang anymore (because of this if/elseif/elseif block). Is this intended? Or can this be changed?

Reply:
I tried `CC=clang CXX=clang++ cmake -G Ninja -DUSE_TSAN=1 ~/workspace/foundationdb` and it chose lld. I think that logic is controlled elsewhere